File: frontend/src/api/inbox.ts
import api from "./scans";

export interface InboxItem {
  id: string;
  kind: string;
  status: string;
  recipient_masked: string;
  attempts: number;
  created_at: string;
  job_id?: string | null;
}

export interface InboxListResponse {
  items: InboxItem[];
  total: number;
}

export async function getInbox(params: {
  page?: number;
  page_size?: number;
  kind?: string;
  status?: string;
}): Promise<InboxListResponse> {
  const { data } = await api.get<InboxListResponse>("/api/inbox", { params });
  return data;
}

export const inboxApi = {
  list: getInbox,
};

File: backend/app/schemas/inbox.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class InboxItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    kind: str
    status: str
    recipient_masked: str
    attempts: int
    created_at: datetime
    job_id: uuid.UUID | None = None


class InboxListResponse(BaseModel):
    items: list[InboxItem]
    total: int
